### ISSUE TITLE
Rename clients

### ISSUE THREAD

**Issue by vmarkovtsev** [bblfsh/documentation#231]
We've got `*-driver` and `client-*` repositories, which are not uniformly named. I suggest to rename client-python, client-go, and client-scala to python-client, go-client, and scala-client.

**TODO:**
- [x] go-client
- [x] python-client
- [x] scala-client https://github.com/bblfsh/client-scala/issues/86
- [ ] update documentation

**Comment by dennwc** [comment]
For the Go client, this will change an import path and may break users. I propose to synchronize this change with a migration to Go modules.

Specifically, move the repository to `go-client`, and change import paths from `gopkg.in` to the Github (`github.com/bblfsh/go-client`). Then, fork the old commit, rename the fork to `client-go` (current name) and archive it.

This way we won't break any clients, and the fork will preserve a link to the new repository.

**Comment by dennwc** [comment]
@juanjux What will be the migration path for Python, if any?

**Comment by creachadair** [comment]
I thought we distributed most of our Go modules via gopkg.in, in which case would it not suffice to update the forwarding? https://help.github.com/articles/renaming-a-repository/ suggests they will issue redirects for the old names.

**Comment by dennwc** [comment]
@creachadair Sure, we can just redirect, but there is no point in distributing packages with `gopkg.in` anymore.

And since we will need to switch from `gopkg.in` sooner or later, which will require to change an import path, I propose to do it now.

The way I described should be less disruptive - old versions of Go can continue using an old client via `gopkg.in`, and 1.11+ can use the new one with modules enabled.

**Comment by juanjux** [comment]
For Python the imported module is just called "bblfsh", and is also the name of the pip module on pypy so nothing should break (last famous words...).

**Comment by bzz** [comment]
Guys, I hate to -1, but please help me understand if this does block something or is a part of some stabilization effort?

I mean, it's kind of a lot of work updating everything - ofc it's nice to have it uniformly named, nobody argues against that, but thinking about current priorities, it may be even more nicer to close all the functional issue that ML and DR teams have put on our backlog first... Or spend time cleaning the path to UASTv2 migration... Or.. or am I missing something here?

**Comment by creachadair** [comment]
> Guys, I hate to -1, but please help me understand if this does block something or is a part of some stabilization effort?
> 
> I mean, it's kind of a lot of work updating everything - ofc it's nice to have it uniformly named, nobody argues against that, but thinking about current priorities, it may be even more nicer to close all the functional issue that ML and DR teams have put on our backlog first... Or spend time cleaning the path to UASTv2 migration... Or.. or am I missing something here?

IMO this is less important than stability and correctness issues, and is not urgent. However, it falls under the rubric of improving the experience for developers using bblfsh as a component, so I think it's worth keeping. I do not think anything is blocked by it, however, so I think it can wait for a rainy day.

**Comment by dennwc** [comment]
Go client was renamed: https://github.com/bblfsh/go-client

**Comment by dennwc** [comment]
Done for Python as well: https://github.com/bblfsh/python-client
One more to go.

**Comment by vmarkovtsev** [comment]
Yes! :muscle:

**Comment by bzz** [comment]
After https://github.com/bblfsh/client-scala/pull/96 is merged I'll rename GH repo for Scala and we can close this.

**Comment by bzz** [comment]
https://github.com/bblfsh/scala-client is the thing, all clients are renamed.

**Comment by bzz** [comment]
Re-opening as documentation update pending

**Comment by bzz** [comment]
Everything is up to date now.
